doc/src/hexdoc_hexcasting/_templates/render.js.jinja

```diff
@@ -0,0 +1,371 @@
+
+    if (!open) {
+      collapsible.addEventListener("toggle", load_func, {once:true})
+    }
+    
+    for (let pattern of patterns) {
+      let attr_to_transfer = ["data-start", "data-string", "width", "height", "data-per-world", "class"];
+      let canvas = document.createElement("canvas");
+      for (const index in attr_to_transfer) {
+        const attr = attr_to_transfer[index];
+        let value = pattern.getAttribute(attr);
+        canvas.setAttribute(attr, value ? value : "");
+      }
+
+      canvas.innerHTML = pattern.innerHTML;
+
+      collapsible.appendChild(canvas);
+
+      if (open) {
+        initializeElem(canvas);
+      }
+
+      pattern.remove()
+    }
+  }
+
+  last_load_func = load_func;
+
+}
+
+/** @type {Point} */
+const intersection_point = {
+  type: "Single",
+  marker: {
+    color: [255, 255, 255, 150],
+    radius: 0.07,
+  }
+};
+
+/** @type {Intersections} */
+const intersections = {
+  type: "EndsAndMiddle",
+  start: {
+    type: "BorderedMatch",
+    match_radius: 0.04,
+    border: intersection_point.marker
+  },
+  middle: intersection_point,
+  end: {
+    type: "Point",
+    point: intersection_point
+  }
+};
+
+const line_thickness = 0.12;
+
+/** @type {Color[]} */
+const colors = [
+  [214, 9, 177, 255],
+  [108, 25, 140, 255],
+  [50, 102, 207, 255],
+  [102, 110, 125, 255],
+];
+
+/** @type {Point} */
+const center_dot = {
+  type: "None"
+};
+
+/** @type {GridOptions} */
+const monocolor = {
+  line_thickness: line_thickness,
+  center_dot: center_dot,
+  pattern_options: {
+    type: "Uniform",
+    lines: {
+      type: "Monocolor",
+      color: colors[0],
+      bent: true
+    },
+    intersections: intersections,
+  },
+};
+
+/** @type {GridOptions} */
+const gradient = {
+  line_thickness: line_thickness,
+  center_dot: center_dot,
+  pattern_options: {
+    type: "Uniform",
+    lines: {
+      type: "Gradient",
+      colors: colors,
+      bent: true,
+      segments_per_color: 15,
+    },
+    intersections: intersections,
+  },
+};
+
+/** @type {GridOptions} */
+const segment = {
+  line_thickness: line_thickness,
+  pattern_options: {
+    type: "Uniform",
+    intersections: intersections,
+    lines: {
+      type: "SegmentColors",
+      colors: colors,
+      triangles: {
+        type: "BorderStartMatch",
+        match_radius: 0.13,
+        border: {
+          color: [255, 255, 255, 255],
+          radius: 0.20,
+        }
+      },
+      collisions: {
+        type: "Dashes",
+        color: [255, 0, 0, 255],
+      }
+    }
+  },
+  center_dot: center_dot,
+}
+
+let selected = "animated";
+
+//this is not programmed to accept the Changing pattern option type (it will crash)
+//it doesn't make sense in this context (as it's rendering single patterns)
+//and it would take more complex color palettes to work properly
+let options = {
+  "monocolor": monocolor,
+  "gradient": gradient,
+  "segment": segment,
+};
```

Comment:
I don't know how to do that (even with the example). Would I just add a placeholder like ```"{{ monocolor_name }}"```, would it be given as an object and I use a path based on LANG? Assuming this involves changing the python side, it might take me a while considering I haven't looked at it too much.